### ISSUE TITLE
[Feature request] Tile contrast

### ISSUE THREAD

**Issue by zvezdochiot** [4lex4/scantailor-advanced#197]
Hi all.

A very favorable effect is the admixture of tile contrast on old scanned pages that have turned gray from time to time.  
Example, original:  
![09](https://user-images.githubusercontent.com/12370082/226103579-1385465c-0220-4bac-b98a-baace91531fe.png)  
ScanTailor Otsu threshold:  
![09 st](https://user-images.githubusercontent.com/12370082/226103608-1c00d395-7824-4d74-9336-9a08bb212071.png)
ScanTailor Wolf threshold:  
![09 st wolf](https://user-images.githubusercontent.com/12370082/226103615-358f7c61-54f7-4108-a145-fdf420b48239.png)

Mixing in 25% tile contrast:  
![09 adc](https://user-images.githubusercontent.com/12370082/226103666-4c423b52-8f74-4ea9-a6a6-6988003dcd02.png)
ScanTailor Otsu threshold:  
![09 adcst](https://user-images.githubusercontent.com/12370082/226103672-d299c64b-6968-448c-a5f5-29c3adccd6b0.png)

I think it's worth looking at ways to reproduce such a contrast without OpenCV: [opencv_adc.cpp.gz](https://github.com/4lex4/scantailor-advanced/files/11008403/opencv_adc.cpp.gz)

This method is much more adjustable in terms of both tile size and mixing ratio than the alternative thresholding methods. And he doesn't have much of an impact on bold headlines like Wolf.

See also:
* https://github.com/josgard94/Image-processing-Algorithm-Contrast-Limited-Adaptative-Histogram-Qualization-CLAHE
* https://github.com/Skymiss/clahe
* https://github.com/llOzzeell/CLAHE-QT

**Comment by zvezdochiot** [comment]
Managed to get a similar effect by simply multiplying an image by itself:  
![1929-22-sq](https://user-images.githubusercontent.com/12370082/226434547-740e9396-03a0-4d55-9cbe-af4ad86d454c.png)

In this case, such a multiplication is again decomposed into an adjustable scheme:  
```
Isq = I * I / 255 => I + I * (I - 255) / 255
```
Add a `k` knob and you get a simple but effective contrast adjustment scheme:  
```
Ik =  I - k * I * (255 - I) / 255, where k = {0.0 .. 1.0}
```

**Comment by zvezdochiot** [comment]
Resolved: [STEX: 0.20230805: control](https://github.com/ImageProcessing-ElectronicPublications/scantailor-experimental/releases/tag/0.20230805)

**Comment by leonleon33156** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
